src/LanguageServer/Impl/Sources/DocumentHighlightSource.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Python.Analysis;
+using Microsoft.Python.Analysis.Documents;
+using Microsoft.Python.Analysis.Modules;
+using Microsoft.Python.Analysis.Types;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.IO;
+using Microsoft.Python.Core.Text;
+using Microsoft.Python.LanguageServer.Documents;
+using Microsoft.Python.LanguageServer.Protocol;
+
+namespace Microsoft.Python.LanguageServer.Sources {
+    internal sealed class DocumentHighlightSource {
+        private const int DocumentHighlightAnalysisTimeout = 10000;
+        private readonly IServiceContainer _services;
+
+        public DocumentHighlightSource(IServiceContainer services) {
+            _services = services;
+        }
+
+        public async Task<DocumentHighlight[]> DocumentHighlightAsync(Uri uri, SourceLocation location, CancellationToken cancellationToken = default) {
+            if (uri != null) {
```

Comment:
Style nit: I'd just do `if (uri == null)` and then return an empty array, rather than having the entire function be a single indented block.

I think the file you based it one does it, but it should probably be changed too (in another PR).

Reply:
This was a copy/paste issue. I have refactored the code now.